console/console.c

```diff
@@ -1586,7 +1593,7 @@ DoCmds(char *master, char *pports, int cmdi)
     char *pcopy;
     char *serverName;
 #if HAVE_GSSAPI
-    int toksize;
+    size_t toksize;
```

Comment:
This change looks correct, because `CanGetGSSContext` either returns `0` on error or `mytok.length` which is `gss_buffer_desc.length` which has type `size_t`. But `CanGetGSSContext` returns that value as `int`, so a large `mytok.length` could overflow `int` and result in an incorrect value for `toksize`.

It seems to me that `CanGetGSSContext` itself should also be changed to return `size_t`, consistent with the `gss_buffer_desc.length` value it returns, and consistent with this change to `toksize` that consumes the return value.

Reply:
Thanks. I meant to change that and forgot before pushing this update. Fixed now.

Reply:
The buffer size is practically no larger than 64Kb.

Reply:
The C standard allows a 16-bit `int` so 64k would overflow if this code is ever used on such a platform (admittedly that might be unlikely).

Reply:
I added a bit more logic to cut off the token size to 64Kb and warn on a client side about upcoming failing authentication. This should help discovering too large tokens in future and will help protecting the server side too.